### PR TITLE
added auth manual

### PR DESCRIPTION
**Pull request by woutslakhorst**

### CODE REVIEW

File: mini-manuals/7-authentication.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
+# Authentication
+
+## Description
+
+This manual describes how to authenticate a user. Nuts authentication makes use of a personal cryptographic means.
+Several means are supported. This manual will describe how to use the dummy means.
+The authentication result is used in the access token request for resources that require user context.
+The authentication result is in the form of a verifiable presentation. 
+It contains enough information for the required logging.
```
Comment (reinkrul) [inline mini-manuals/7-authentication.md:9]: required by?

File: mini-manuals/7-authentication.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+## Resources
+
+- **IRMA getting started guide**: https://nuts-node.readthedocs.io/en/latest/pages/getting-started/5-irma-contract.html
+- **Auth APIs**: https://nuts-node.readthedocs.io/en/latest/pages/api.html?urls.primaryName=Auth
+- **Authentication token RFC**: https://nuts-foundation.gitbook.io/drafts/rfc/rfc002-authentication-token
+
+## Compose a contract
+
+User authentication is done by signing a contract. 
+This contract is shown on the means of the user.
```
Comment (reinkrul) [inline mini-manuals/7-authentication.md:20]: "shown on the means of the user"? (weird sentence)

"device"?

File: mini-manuals/7-authentication.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+It contains additional information like the name of the organization and the validity of the contract.
+Contracts are formatted, so they can also be easily parsed.
+
+A login contract can be formatted by calling:
```
Comment (reinkrul) [inline mini-manuals/7-authentication.md:24]: formatted -> composed

File: mini-manuals/7-authentication.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  "type": "PractitionerLogin",
+  "language": "EN",
+  "version": "v3",
+  "legalEntity": "did:nuts:Ft8NRLzSjxyw8AmTHVtJ9ehBctXpsaQjshmHnqWCATEz",
+  "validFrom": "2021-06-24T14:32:00+02:00",
+  "validDuration": "1h"
+}
+```
+
+The Nuts node will try to find a trusted organization credential for the given DID. The name and place of that credential will be used in the contract.
```
Comment (reinkrul) [inline mini-manuals/7-authentication.md:40]: place -> city (that's how it's called in the VC)

File: mini-manuals/7-authentication.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+Which returns something like:
+
+```
+{
+    "status": "pending"
+}
+```
+
+When the signing session has not completed yet a means specific status is returned like: *pending* or *in-progress*.
```
Comment (reinkrul) [inline mini-manuals/7-authentication.md:100]: Maybe mention this endpoint is normally polled for status updates. That the dummy means has to be called 3 times is ofc somewhat arbitrary, so not 100% whether people should base their implementation on this.

File: mini-manuals/7-authentication.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+{
+    "status": "completed",
+    "verifiablePresentation": {
+        ...
+    }
+}
+```
+
+The `status` has changed to completed and a `verifiablePresentation` has been added.
+The contents of that field can be used when requesting an access token. It has to be base64 encoded and entered in the `identity` field of the `/internal/auth/v1/request-access-token` request body.
```
Comment (reinkrul) [inline mini-manuals/7-authentication.md:114]: > The contents of the `verifiablePresentation` field can be used

(makes it more explicit which field is meant)

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: mini-manuals/7-authentication.md (hunk 1 above)
Comment (woutslakhorst) [reply mini-manuals/7-authentication.md:9]: added NEN7513

File: mini-manuals/7-authentication.md (hunk 5 above)
Comment (woutslakhorst) [reply mini-manuals/7-authentication.md:100]: replace it with "polling untill completed"

**Review by reinkrul — APPROVED** [review]

**Pull request merged**